apis/notification/v1alpha1/snssubscription_types.go

```diff
@@ -0,0 +1,143 @@
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SNSSubscriptionParameters define the desired state of a AWS SNS Topic
+type SNSSubscriptionParameters struct {
+
+	// TopicArn is the Arn of the SNS Topic
+	// +optional
+	TopicArn string `json:"topicArn,omitempty"`
+
+	// TopicArnRef references a SNS Topic and retrieves its TopicArn
+	// +optional
+	TopicArnRef *runtimev1alpha1.Reference `json:"topicArnRef,omitempty"`
+
+	// TopicArnSelector selects a reference to a SNS Topic and retrieves
+	// its TopicArn
+	// +optional
+	TopicArnSelector *runtimev1alpha1.Selector `json:"topicArnSelector,omitempty"`
+
+	// The subscription's protocol.
+	// +immutable
+	Protocol *string `json:"protocol"`
+
+	// The subscription's endpoint
+	// +immutable
+	Endpoint *string `json:"endpoint"`
+
+	//  DeliveryPolicy defines how Amazon SNS retries failed
+	//  deliveries to HTTP/S endpoints.
+	// +optional
+	DeliveryPolicy *string `json:"deliveryPolicy,omitempty"`
+
+	//  The simple JSON object that lets your subscriber receive
+	//  only a subset of messages, rather than receiving every message published
+	//  to the topic.
+	// +optional
+	FilterPolicy *string `json:"filterPolicy,omitempty"`
+
+	//  When set to true, enables raw message delivery
+	//  to Amazon SQS or HTTP/S endpoints. This eliminates the need for the endpoints
+	//  to process JSON formatting, which is otherwise created for Amazon SNS
+	//  metadata.
+	// +optional
+	RawMessageDelivery *string `json:"rawMessageDelivery,omitempty"`
+
+	//  When specified, sends undeliverable messages to the
+	//  specified Amazon SQS dead-letter queue. Messages that can't be delivered
+	//  due to client errors (for example, when the subscribed endpoint is unreachable)
+	//  or server errors (for example, when the service that powers the subscribed
+	//  endpoint becomes unavailable) are held in the dead-letter queue for further
+	//  analysis or reprocessing.
+	// +optional
+	RedrivePolicy *string `json:"redrivePolicy,omitempty"`
+}
+
+// SNSSubscriptionSpec defined the desired state of a AWS SNS Topic
+type SNSSubscriptionSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  SNSSubscriptionParameters `json:"forProvider"`
+}
+
+// ConfirmationStatus represents Status of SNS Subscription Confirmation
+type ConfirmationStatus string
+
+const (
+	// ConfirmationPending represents Pending Confirmation Request for SNS Subscription
+	ConfirmationPending ConfirmationStatus = "ConfirmationPending"
+	// ConfirmationSuccessful represents confirmed SNS Subscription
+	ConfirmationSuccessful ConfirmationStatus = "Confirmed"
+)
+
+// SNSSubscriptionObservation represents the observed state of a AWS SNS Topic
+type SNSSubscriptionObservation struct {
+
+	// The subscription's owner.
+	// +optional
```

Comment:
I don't think we usually worry too much about the `// +optional` convention for status messages (that are never written by humans). That being said, it doesn't hurt anything to have them there.